.github/workflows/commitlint.yml

```diff
@@ -0,0 +1,40 @@
+# Run commitlint on the commit messages in a pull request.
+
+name: Lint Commit Messages
+
+on:
+  # This action is meant to be used from other actions.
+  # https://docs.github.com/en/actions/learn-github-actions/reusing-workflows
+  - workflow_call
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-latest
```

Comment:
we've been somewhat explicit in workflows everywhere and specified Ubuntu 20.04, but I think this workflow won't have that big an impact if it switches to a newer version of Ubuntu